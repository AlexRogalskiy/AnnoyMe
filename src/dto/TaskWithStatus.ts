import Task from "./Task";

/** Task representation with current status. */
export default interface TaskWithStatus extends Task {
  isActive: boolean;
}
