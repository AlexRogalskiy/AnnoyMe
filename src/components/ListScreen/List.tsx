import React from "react";
import { ScrollView } from "react-native";
import TaskWithStatus from "../../dto/TaskWithStatus";
import Item from "./Item";
import styles from "./styles";

interface Props {
  tasks: TaskWithStatus[];

  onItemPress: (task: TaskWithStatus) => void;
}

/** Scrollable list of tasks. */
export default ({ tasks, onItemPress }: Props) => (
  <ScrollView contentContainerStyle={styles.container}>
    {tasks.map(task => (
      <Item
        task={task}
        key={`task-item-${task.id}`}
        onPress={() => onItemPress(task)}
        testID={`task-item-${task.id}`}
      />
    ))}
  </ScrollView>
);
