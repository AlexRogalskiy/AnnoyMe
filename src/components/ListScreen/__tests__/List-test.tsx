import React from "react";
import renderer, { act } from "react-test-renderer";
import TaskWithStatus from "../../../dto/TaskWithStatus";
import List from "../List";

const tasks = [
  {
    id: "1",
    created: new Date(),
    title: "First test task",
    schedule: {},
    isActive: false,
  },
  {
    id: "2",
    created: new Date(),
    title: "Second test task",
    schedule: {},
    isActive: true,
  },
];

it("List renders all tasks", () => {
  const tree = renderer
    .create(
      <List tasks={tasks} onItemPress={(_: TaskWithStatus) => undefined} />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot("List-all");
});

it("List handles item clicks", () => {
  const onItemPress = jest.fn((_: TaskWithStatus) => undefined);

  const component = renderer.create(
    <List tasks={tasks} onItemPress={onItemPress} />,
  ).root;

  const secondTask = component.findByProps({ testID: "task-item-2" });
  expect(secondTask).toBeTruthy();

  act(() => secondTask.props.onPress());

  const firstTask = component.findByProps({ testID: "task-item-1" });
  expect(firstTask).toBeTruthy();

  act(() => firstTask.props.onPress());

  expect(onItemPress.mock.calls).toEqual([[tasks[1]], [tasks[0]]]);
});
