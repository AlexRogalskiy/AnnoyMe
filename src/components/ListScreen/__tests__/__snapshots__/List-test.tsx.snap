// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders all tasks: List-all 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "#eeeeee",
          "borderBottomWidth": 2,
          "opacity": 1,
        }
      }
      testID="task-item"
    >
      <Text
        style={
          Object {
            "fontWeight": "400",
            "padding": 20,
          }
        }
      >
        First test task
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#9ccc65",
          "borderBottomColor": "#eeeeee",
          "borderBottomWidth": 2,
          "opacity": 1,
        }
      }
      testID="task-item"
    >
      <Text
        style={
          Object {
            "fontWeight": "400",
            "padding": 20,
          }
        }
      >
        Second test task
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
